### PR TITLE
New disabled state from properties

### PR DESCRIPTION
**Pull request by rkuykendall**

### CODE REVIEW

File: src/AvailabilityGrid/AvailabilityGrid.jsx [hunk 1]
```diff
@@ -230,21 +231,23 @@ const AvailabilityGrid = React.createClass({
...
-        selectionMode: newMode,
-        selectionStartDay: startDay,
-        selectionStartSlot: startSlot
-      });
-    } else {
-      this.setState({
-        selectionMode: newMode
-      });
+    if (!this.props.disabled) {
+      if (newMode == 'selecting' || newMode == 'unselecting') {
```
Comment (diezguerra) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: I understand the code was there before, but shouldn't you be using triple =?

**Comment by diezguerra** [comment]
lgtm :+1: (except for the css, that's not my thing).

File: src/AvailabilityGrid/AvailabilityGrid.jsx (hunk 1 above)
Comment (rkuykendall) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: From what I can tell, it's exactly the same thing. I could change it though. @evanrs?

Comment (diezguerra) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: that's right, it should be the same thing. I think the JS default is to use the triple one by default

File: src/AvailabilityGrid/availability_grid.less [hunk 2]
```diff
@@ -1,5 +1,26 @@
...
 
+.availability-grid.disabled {
+  color: @gray50;
+
+  .availability-grid-days {
+    color: @gray50;
+
+    .availability-grid-day {
+
+      .availability-grid-slot {
```
Comment (tholex) [inline src/AvailabilityGrid/availability_grid.less]: no need to nest these...

File: src/AvailabilityGrid/availability_grid.less [hunk 3]
```diff
@@ -1,5 +1,26 @@
 @import "~tfstyleguide/vars";
 
+.availability-grid.disabled {
```
Comment (tholex) [inline src/AvailabilityGrid/availability_grid.less]: If you want to "perfectly mimic" the rest of our LESS, use `.availability-grid__disabled` as the modifier class, that way you can actually drop all of these into a block inside `.availability-grid` and still have it be searchable.

Comment (rkuykendall) [inline src/AvailabilityGrid/availability_grid.less]: Awesome, thanks for the help.

File: src/AvailabilityGrid/AvailabilityGrid.jsx (hunk 1 above)
Comment (evanrs) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: @diezguerra is correct you should only ever use strict equals `===`

Comment (evanrs) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: But also favor regular expressions for simple string matches like this
```es6
if (/selecting/g.test(newMode) {
}
```

File: src/AvailabilityGrid/AvailabilityGrid.jsx [hunk 4]
```diff
@@ -306,9 +309,14 @@ const AvailabilityGrid = React.createClass({
       );
     });
 
+    let classes = classNames({
```
Comment (evanrs) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]:
```es6
classNames('availability-grid', {'availability-grid__disabled': this.props.disabled})
```

File: src/AvailabilityGrid/AvailabilityGrid.jsx [hunk 5]
```diff
@@ -306,9 +309,14 @@ const AvailabilityGrid = React.createClass({
...
+      'availability-grid': true,
+      'availability-grid__disabled': this.props.disabled
+    });
+
     return (
       <div
-          className="availability-grid"
+          className={classes}
           onMouseEnter={this.handleMouseEnter}
```
Comment (evanrs) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: Inconsistent naming, either handleMouseEnter or onMouseEnter, @ripleyaffect preferes `_handleMouseEnter`, however on is fewer characters and just as transparent.

File: src/AvailabilityGrid/AvailabilityGrid.jsx (hunk 1 above)
Comment (rkuykendall) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: Wait... what? Why would you do that? Regular expressions for simple string comparisons?

File: src/AvailabilityGrid/AvailabilityGrid.jsx (hunk 5 above)
Comment (rkuykendall) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: Hm, I actually thought `handle` was what you called yours and `on` was what you passed down as a param. Not inconsistent, just clearer.

File: src/AvailabilityGrid/AvailabilityGrid.jsx (hunk 4 above)
Comment (rkuykendall) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: Fixed on https://github.com/Thinkful/ui/pull/98

**Pull request merged**